src/equations/ideal_mhd_multiion_2d.jl

```diff
@@ -118,12 +118,9 @@ function initial_condition_weak_blast_wave(x, t, equations::IdealMhdMultiIonEqua
   v2 = r > 0.5 ? 0.0 : 0.1882 * sin(phi)
   p = r > 0.5 ? 1.0 : 1.245
 
-  #prim = (0.01, 0.01, 0.01)
   prim = (1.0, 1.0, 1.0)
   for i in eachcomponent(equations)
-    prim = (prim..., 2^(i-1) * (1-2)/(1-2^ncomponents(equations)) * rho, v1, v2, 0.0, p)
-    #prim = (prim..., rho, v1, 0.0, 0.0, p)
-    #prim = (prim..., 1.0, 1.0, 0.0, 0.0, 100.0)
+    prim = (prim..., 2^(i-1) * (1-2)/(1-2^ncomponents(equations)) * rho, v1, v2, zero(real(equations)), p)
```

Comment:
This is also (probably) type unstable, like all the other tuple extensions in a loop before. Thus you should init `prim = zero(MVector...)` above and then just fill the values here as you did elsewhere

Reply:
Makes sense. I will change them in `prim2cons`, `cons2prim`, and `cons2entropy`. However, I don't know how to change `varnames`. I tried a similar strategy with `zero(MVector{nvariables(equations), Char})`, but that didn't work. Do you know how to do it for strings/characters?

Reply:
I get the following:
```julia
julia> using StaticArrays

julia> varnames=MVector{3, String}
MVector{3, String} (alias for MArray{Tuple{3}, String, 1, 3})

julia> varnames[1]="rho"
ERROR: MethodError: no method matching setindex!(::Type{MVector{3, String}}, ::String, ::Int64)
Stacktrace:
 [1] top-level scope
   @ REPL[6]:1

julia> varnames[1]="B"
ERROR: MethodError: no method matching setindex!(::Type{MVector{3, String}}, ::String, ::Int64)
Stacktrace:
 [1] top-level scope
   @ REPL[7]:1

julia> varnames=MVector{3, Char}
MVector{3, Char} (alias for MArray{Tuple{3}, Char, 1, 3})

julia> varnames[1]="B"
ERROR: MethodError: no method matching setindex!(::Type{MVector{3, Char}}, ::String, ::Int64)
Stacktrace:
 [1] top-level scope
   @ REPL[9]:1
```

Reply:
Right. Yeah, give it a try.

Reply:
Ok, I modified ^prim2cons`, `cons2prim`, and `cons2entropy` in 968b688. `varnames` will remain the same for the moment.